sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileApiTests.java

```diff
@@ -637,6 +640,99 @@ public void downloadRetryDefault() {
         assertEquals(bodyStr, DATA.getDefaultText());
     }
 
+    @Test
+    public void downloadToFileWithPartialDownloads() throws Exception {
+        File uploadFile = File.createTempFile(CoreUtils.randomUuid().toString(), ".txt");
+        uploadFile.deleteOnExit();
+        Files.write(uploadFile.toPath(), DATA.getDefaultBytes());
+
+        File outFile = new File(generatePathName() + ".txt");
+        if (outFile.exists()) {
+            assertTrue(outFile.delete());
+        }
+        outFile.deleteOnExit();
+
+        MockPartialResponsePolicy policy = new MockPartialResponsePolicy(5);
+
+        // Create a ShareFileClient for download using the custom pipeline
+        ShareFileClient downloadClient
+            = getFileClient(ENVIRONMENT.getPrimaryAccount().getCredential(), primaryFileClient.getFileUrl(), policy);
+
+        // Upload the test data
+        primaryFileClient.create(DATA.getDefaultDataSize());
+        primaryFileClient.uploadFromFile(uploadFile.toString());
+
+        downloadClient.downloadToFile(outFile.toString());
+
+        byte[] downloadedData = Files.readAllBytes(outFile.toPath());
+        Assertions.assertArrayEquals(DATA.getDefaultBytes(), downloadedData);
+
+        // Assert that we retried the correct number of times (5)
+        assertEquals(0, policy.getTriesRemaining());
+
+        // Assert that the range headers that were retried match what was returned from MockPartialResponsePolicy
+        List<String> expectedRanges = expectedHeaderRanges();
+        List<String> actualRanges = policy.getRangeHeaders();
+        assertEquals(expectedRanges.size(), actualRanges.size());
+        for (int i = 0; i < actualRanges.size(); i++) {
+            assertEquals(expectedRanges.get(i), actualRanges.get(i));
+        }
```

Comment:
When asserting the actual and expected range header lists in each of these test cases, can we just directly assert that they are equal? e.g.:
```java
assertEquals(actualRanges, expectedRanges);
```